### PR TITLE
Make `str()` work for non-constant values

### PR DESCRIPTION
**Pull request by Abdelrahman-Kh-Fouad**
Related to #761 .
I have missed `ord()` on converting line.
there was a bug in reversing loop.

### CODE REVIEW

**Comment by Abdelrahman-Kh-Fouad** [comment]
@certik @namannimmo10 
ready for reviewing.

File: src/runtime/lpython_builtin.py [hunk 1]
```diff
@@ -135,11 +135,11 @@ def str(x: i32) -> str:
     rev_result_len: i32
     rev_result_len = 0
     while x > 0:
-        rev_result += chr(x - _lpython_floordiv(x, 10)*10)
+        rev_result += chr(ord('0') + (x - _lpython_floordiv(x, 10) * 10))
```
Comment (namannimmo10) [inline src/runtime/lpython_builtin.py:138]:
```suggestion
        rev_result += chr(48 + (x - _lpython_floordiv(x, 10) * 10))
```

**Comment by namannimmo10** [comment]
To make sure that this works with LLVM, please rebase with `main`.

**Comment by namannimmo10** [comment]
Thanks. Can you also make `str()` work for floats and other inputs?

**Pull request merged**